manifests/certonly.pp

```diff
@@ -106,27 +111,38 @@
     $command_end = join(['',] + $additional_args, ' ')
   }
 
-  $command = "${command_start}${command_domains}${command_end}"
-  $live_path_domain = regsubst($domains[0], '^\*\.', '')
-  $live_path = "${config_dir}/live/${live_path_domain}/cert.pem"
+  $live_path = "${config_dir}/live/${title}/cert.pem"
 
   $execution_environment = [ "VENV_PATH=${letsencrypt::venv_path}", ] + $environment
   $verify_domains = join(unique($domains), ' ')
+
+  if $ensure == 'present' {
+    $exec_params = {
+      'command' => "${command_start}${command_domains}${command_end}",
+      'unless'  => "/usr/local/sbin/letsencrypt-domain-validation ${live_path} ${verify_domains}",
+    }
+  } else {
+    $exec_params = {
+      'command' => "${letsencrypt_command} --text --agree-tos --non-interactive delete --cert-name ${title}",
+      'onlyif'  => "/usr/local/sbin/letsencrypt-domain-validation ${live_path} ${verify_domains}",
+    }
+  }
+
   exec { "letsencrypt certonly ${title}":
```

Comment:
We should be able to forge `$command` yet in a generic manner.

At line 70 :
https://github.com/voxpupuli/puppet-letsencrypt/blob/e9808ad59a585cea2fb67d72b774104cbc843ed2/manifests/certonly.pp#L70
```
case $ensure {
  'present': {
    if ($custom_plugin) {
      $command_start = "${letsencrypt_command} --text --agree-tos --non-interactive certonly --rsa-key-size ${key_size} "
    } else {
      $command_start = "${letsencrypt_command} --text --agree-tos --non-interactive certonly --rsa-key-size ${key_size} -a ${plugin} "
    }
  } 
  'absent': {
    $command_start = "${letsencrypt_command} --text --agree-tos --non-interactive delete "
  } 
  default: {
    fail ('unexpected value of ensure')
  } 
}
```


default at line 100
https://github.com/voxpupuli/puppet-letsencrypt/blob/e9808ad59a585cea2fb67d72b774104cbc843ed2/manifests/certonly.pp#L100
```
default: {
  case $ensure {
    'present': {
      $_command_domains = join($domains, ' -d ')
      $command_domains  = "--cert-name ${title} -d ${_command_domains}"
    }
    'absent': {
      $command_domains = "--cert-name ${title}"
    }
    default: {
      fail ('unexpected value of ensure')
    }
  } 
} 
```

Reply:
Is there any preference for
```
case $ensure {
  'present': {}
  'absent': {}
  default: {}
}
```
over
```
case $ensure {
  'present': {}
  default: {}
}
```
?

Since $ensure is an Enum that may only be 'present' or 'absent' it theoretically could never hit the default clause.

Reply:
Or a third option;
```
if $ensure == 'present' {} else {}
```

Reply:
I'd probably go with option 3.

Reply:
Job done, and all the tests are still happy.